File: tmp/TestAndroid/common/src/main/java/com/android/launcher3/common/base/BasePresenter.java
package com.android.launcher3.common.base;

import java.lang.ref.Reference;
import java.lang.ref.WeakReference;

/**
 * Created by shensonglong on 2018/1/28.
 */
public class BasePresenter<V> {

    protected final String TAG = BasePresenter.class.getSimpleName() + "--->>>";

    protected Reference<V> mViewRef;
    protected V mView;

    public void attachView(V view) {
        mViewRef = new WeakReference<V>(view);
        mView = mViewRef.get();
    }

    public V getView() {
        if (mViewRef != null) {
            return mViewRef.get();
        }
        return null;
    }

    public boolean isViewAttached() {
        return getView() != null;
    }

    public void detachView() {
        if (mViewRef != null) {
            mViewRef.clear();
            mViewRef = null;
        }
    }

}

File: head_layout/app/src/main/java/com/baehug/callui/base/BasePresenter.kt
package com.baehug.callui.base

/**
 * Author : yanyong
 * Date : 2024/8/17
 * Details :
 */
open class BasePresenter<T : BaseView> : BasePre<T> {
    private  var mView: T? = null

    override fun detachView() {
        mView = null
    }

    override fun attachView(view: T?) {
        mView = view
    }
}
